support-frontend/app/controllers/Application.scala

```diff
@@ -274,15 +281,29 @@ class Application(
   }
 
   def router(countryGroupId: String): Action[AnyContent] = MaybeAuthenticatedAction { implicit request =>
-    implicit val settings: AllSettings = settingsProvider.getAllSettings()
+    request.queryString
+      .getOrElse("product", Nil)
+      .headOption
+      .flatMap(productString => Product.fromString(productString))
+      .map(routeForProduct(_))
+      .getOrElse(BadRequest("No product name provided"))
+  }
```

Comment:
[We're doing this validation on the frontend](https://github.com/guardian/support-frontend/blob/main/support-frontend/assets/pages/%5BcountryGroupId%5D/checkout.tsx#L321-L337).

The point of this was to move as much logic to the client as possible to avoid having to context switch. I don't think this is much of an issue, just something to be mindful of.